src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPassBase.cs

```diff
@@ -58,12 +58,24 @@ protected async Task<List<TextChange>> AdjustIndentationAsync(FormattingContext
 #if DEBUG
             var spanText = context.SourceText.GetSubText(mappingSpan).ToString();
 #endif
-            if (!ShouldFormat(context, mappingSpan, allowImplicitStatements: true))
+            if (!ShouldFormat(context, mappingSpan, allowImplicitStatements: true, out var owner))
             {
                 // We don't care about this range as this can potentially lead to incorrect scopes.
                 continue;
             }
 
+            // Special case: We are looking for "significant locations" that should affect indentation, but
+            // single line explicit expressions shouldn't. If there are any at the start of a line, the next
+            // loop will find them. Sadly the ShouldFormat method is used in too many places, for too many
+            // different purposes, to put this check there.
+            if (owner is not null &&
```

Comment:
Nit: I'd there a way we can avoid walking up all ancestors?

Reply:
Agreed. It'd be nice if we could stop the iteration.

Reply:
Imagine a list of crimes 4 times longer than the longest CVS receipt you've ever gotten. This is adding one line item at the end of that :P

I can have a play with this spot specifically, but a lot of formatting has to look up the tree like this because Razor can have C# in Html in C# etc. so you can be at a point of C# mapping where you are not direcitly inside a certain expression type, but if you keep looking up, you are.

Reply:
This one could stop when it finds an element that is beyond a single line.

It is a nit, and agree that it likely is not noticeable for perf. Definitely not blocking

Reply:
Decided to fix this, and if there is some weird case with nested whatsits that it misses, then we'll hear about it. The whole thing was broken before anyway :)